aten/src/ATen/TensorSubclassLikeUtils.h

```diff
@@ -66,7 +66,9 @@ inline bool areAnyOptionalTensorSubclassLike(
   if (c10::impl::dispatch_mode_enabled())
     return true;
   return std::any_of(
-      tensors.begin(), tensors.end(), [](const optional<Tensor>& opt_tensor) {
+      tensors.begin(),
+      tensors.end(),
+      [](const c10::optional<Tensor>& opt_tensor) {
```

Comment:
add `c10::` to `optional<Tensor>&` as compiler was unable to resolve between `std::optional` and `c10::optional`.

Reply:
We are deprecating c10:optional, Does `std::optional` work here?

Reply:
I think it would, will update this. Thanks!

Reply:
I would keep c10::optional here, because std::optional can and has caused us build problems that need to be resolved. As a part of the migration I expect us to make c10::optional an alias for std::optional which would resolve this.